crates/sui/src/faucet.rs

```diff
@@ -0,0 +1,199 @@
+        .route("/v1/gas", post(faucet_batch_request))
+        .route("/v1/status/:task_id", get(request_status))
+        .layer(
+            ServiceBuilder::new()
+                .layer(cors)
+                .layer(Extension(app_state))
+                .into_inner(),
+        );
+
+    let addr = SocketAddr::from(([127, 0, 0, 1], port));
+
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await?;
+
+    Ok(())
+}
+
+/// basic handler that responds with a static string
+async fn health() -> &'static str {
+    "OK"
+}
+
+/// Requests a SUI token from the faucet.
+async fn faucet_request(
+    Extension(state): Extension<Arc<AppState>>,
+    Json(payload): Json<FaucetRequest>,
+) -> impl IntoResponse {
+    let result = match payload {
+        FaucetRequest::FixedAmountRequest(FixedAmountRequest { recipient }) => {
+            state.faucet.request_sui_coins(recipient).await
+        }
+        _ => {
+            return (
+                StatusCode::BAD_REQUEST,
+                Json(FaucetResponse::from(FaucetError::Internal(
+                    "Input Error.".to_string(),
+                ))),
+            )
+        }
+    };
+
+    if !result.transferred_gas_objects.is_empty() {
+        (StatusCode::CREATED, Json(result))
+    } else {
+        (StatusCode::INTERNAL_SERVER_ERROR, Json(result))
+    }
+}
+
+/// Make a request to the faucet that will be batched with other rqeuests.
+async fn faucet_batch_request(
+    Extension(state): Extension<Arc<AppState>>,
+    Json(payload): Json<FaucetRequest>,
+) -> impl IntoResponse {
+    let result = match payload {
+        FaucetRequest::FixedAmountRequest(FixedAmountRequest { recipient }) => {
+            state.faucet.batch_request_sui_coins(recipient).await
+        }
+        _ => {
+            return (
+                StatusCode::BAD_REQUEST,
+                Json(BatchFaucetResponse::from(FaucetError::Internal(
+                    "Input Error.".to_string(),
+                ))),
+            )
+        }
+    };
+    if result.task.is_some() {
+        (StatusCode::CREATED, Json(result))
+    } else {
+        (StatusCode::INTERNAL_SERVER_ERROR, Json(result))
+    }
+}
+
+async fn request_status(
+    Extension(state): Extension<Arc<AppState>>,
+    Path(id): Path<String>,
+) -> impl IntoResponse {
+    match Uuid::parse_str(&id) {
+        Ok(task_id) => {
+            let status = state.faucet.get_batch_send_status(task_id).await;
+            (StatusCode::CREATED, Json(status))
+        }
+        Err(e) => (
+            StatusCode::INTERNAL_SERVER_ERROR,
+            Json(BatchStatusFaucetResponse::from(FaucetError::Internal(
+                e.to_string(),
+            ))),
+        ),
+    }
+}
+
+/// Faucet requires its own wallet context in order to send SUI to addresses.
+/// This functions initializes a wallet context in a temporary directory, and returns the object.
+pub fn new_wallet_context_for_faucet(
```

Comment:
I think that this function was not present in the old faucet wrapper - why do we need it now?

Reply:
Great question. The `sui-test-validator` used the `LocalNewCluster` from `sui-cluster-test` crate, to create a local cluster, and then that object is passed to the [FaucetClientFactory](https://github.com/MystenLabs/sui/blob/main/crates/sui-cluster-test/src/faucet.rs#L20) to create a new faucet. In this [`new_from_cluster` method](https://github.com/MystenLabs/sui/blob/main/crates/sui-cluster-test/src/faucet.rs#L21), the `new_wallet_context_from_cluster` is called, which is a [function implemented ](https://github.com/MystenLabs/sui/blob/main/crates/sui-cluster-test/src/cluster.rs#L337C8-L337C39)for the `Cluster` type. 

The code in this PR is using directly the `SwarmBuilder` as in the original `sui start` code for a few reasons, but one of them being related to the `data_ingestion_dir` from above which I did not manage to make it work properly with the `LocalNewCluster`.